### PR TITLE
add bandit scanning to ultratest and travis

### PR DESCRIPTION
**Pull request by jseppi**
This adds `bandit` to both ultratest and to our CI tests. 

Ready for review, @toolness 

Related: https://github.com/18F/calc/pull/405 and https://github.com/18F/Infrastructure/issues/649 (our ATO issue)

### CODE REVIEW

File: .bandit [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[bandit]
+exclude: node_modules,selenium_tests,qunit_tests,api/tests,meta/management/commands,docker_django_management.py
```
Comment (jseppi) [inline .bandit]: I created this exclude list based on files that bandit found issues, but that were false positives. Tests and custom commands will never be run on production.

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +1,4 @@
+bandit>=1.0,<2.0
```
Comment (jseppi) [inline requirements.txt]: I gave bandit a version range because I think we can safely, and would want to, have it get upgraded as new releases come out.

File: .bandit (hunk 1 above)
Comment (jseppi) [inline .bandit]: <details>
<summary>Results without tests and commands ignored</summary>
```sh
$ bandit -r . -x node_modules
[main]	INFO	profile include tests: None
[main]	INFO	profile exclude tests: None
[main]	INFO	cli include tests: None
[main]	INFO	cli exclude tests: None
[main]	INFO	running on Python 3.4.3
71 [0.. 50.. ]
Run started:2016-07-26 14:40:22.287649

Test results:
>> Issue: [B311:blacklist] Standard pseudo-random generators are not suitable for security/cryptographic purposes.
   Severity: Low   Confidence: High
   Location: ./api/tests/test_utils.py:37
36	    def test_works_on_decimal_values(self):
37	        values = [decimal.Decimal(random.randrange(1, 101, 1))
38	                  for _ in range(10000)]

--------------------------------------------------
>> Issue: [B311:blacklist] Standard pseudo-random generators are not suitable for security/cryptographic purposes.
   Severity: Low   Confidence: High
   Location: ./api/tests/test_utils.py:55
54	        num_bins = 12
55	        values = [round(random.random(), 2) for _ in range(10000)]
56	        bins = get_histogram(values, num_bins)

--------------------------------------------------
>> Issue: [B404:blacklist] Consider possible security implications associated with subprocess module.
   Severity: Low   Confidence: High
   Location: ./docker_django_management.py:70
69	import signal
70	import subprocess
71
72
73	MY_DIR = os.path.abspath(os.path.dirname(__file__))

--------------------------------------------------
>> Issue: [B602:subprocess_popen_with_shell_equals_true] subprocess call with shell=True identified, security issue.
   Severity: High   Confidence: High
   Location: ./docker_django_management.py:105
104	            "ps --ppid=%d -o pid | awk 'NR>1' | xargs echo" % os.getpid(),
105	            shell=True
106	        )
107	        return map(int, output.split())
108

--------------------------------------------------
>> Issue: [B606:start_process_with_no_shell] Starting a process without a shell.
   Severity: Low   Confidence: Medium
   Location: ./docker_django_management.py:192
191	        try:
192	            os.execvp('docker-compose', [
193	                'docker-compose', 'run', CONTAINER_NAME, 'python'
194	            ] + argv)
195	        except OSError:

--------------------------------------------------
>> Issue: [B607:start_process_with_partial_path] Starting a process with a partial executable path
   Severity: Low   Confidence: High
   Location: ./docker_django_management.py:192
191	        try:
192	            os.execvp('docker-compose', [
193	                'docker-compose', 'run', CONTAINER_NAME, 'python'
194	            ] + argv)
195	        except OSError:

--------------------------------------------------
>> Issue: [B603:subprocess_without_shell_equals_true] subprocess call - check for execution of untrusted input.
   Severity: Low   Confidence: High
   Location: ./docker_django_management.py:242
241	            home_dir = '/home/%s' % username
242	            subprocess.check_call([
243	                'useradd',
244	                '-d', home_dir,
245	                '-m', username,
246	                '-u', str(HOST_UID)
247	            ])

--------------------------------------------------
>> Issue: [B607:start_process_with_partial_path] Starting a process with a partial executable path
   Severity: Low   Confidence: High
   Location: ./docker_django_management.py:242
241	            home_dir = '/home/%s' % username
242	            subprocess.check_call([
243	                'useradd',
244	                '-d', home_dir,
245	                '-m', username,
246	                '-u', str(HOST_UID)
247	            ])

--------------------------------------------------
>> Issue: [B606:start_process_with_no_shell] Starting a process without a shell.
   Severity: Low   Confidence: Medium
   Location: ./docker_django_management.py:250
249	        os.setuid(HOST_UID)
250	    os.execvp(argv[1], argv[1:])
251

--------------------------------------------------
>> Issue: [B404:blacklist] Consider possible security implications associated with subprocess module.
   Severity: Low   Confidence: High
   Location: ./meta/management/commands/ultratest.py:3
2	import tempfile
3	import subprocess
4	import djclick as click

--------------------------------------------------
>> Issue: [B602:subprocess_popen_with_shell_equals_true] subprocess call with shell=True identified, security issue.
   Severity: High   Confidence: High
   Location: ./meta/management/commands/ultratest.py:125
124	            result = subprocess.call(
125	                entry['cmd'], stdout=out, stderr=subprocess.STDOUT, shell=True
126	            )
127

--------------------------------------------------
>> Issue: [B404:blacklist] Consider possible security implications associated with subprocess module.
   Severity: Low   Confidence: High
   Location: ./qunit_tests/tests.py:2
1	import os
2	import subprocess
3
4	from django.test import LiveServerTestCase

--------------------------------------------------
>> Issue: [B603:subprocess_without_shell_equals_true] subprocess call - check for execution of untrusted input.
   Severity: Low   Confidence: High
   Location: ./qunit_tests/tests.py:16
15	    def test_qunit(self):
16	        subprocess.check_call([
17	            'phantomjs', RUNNER_PATH, self.live_server_url + '/tests/'
18	        ])

--------------------------------------------------
>> Issue: [B607:start_process_with_partial_path] Starting a process with a partial executable path
   Severity: Low   Confidence: High
   Location: ./qunit_tests/tests.py:16
15	    def test_qunit(self):
16	        subprocess.check_call([
17	            'phantomjs', RUNNER_PATH, self.live_server_url + '/tests/'
18	        ])

--------------------------------------------------
>> Issue: [B404:blacklist] Consider possible security implications associated with subprocess module.
   Severity: Low   Confidence: High
   Location: ./selenium_tests/utils.py:4
3
4	import subprocess
5
6
7	_static_assets_built = False

--------------------------------------------------
>> Issue: [B603:subprocess_without_shell_equals_true] subprocess call - check for execution of untrusted input.
   Severity: Low   Confidence: High
   Location: ./selenium_tests/utils.py:22
21	    if not _static_assets_built:
22	        subprocess.check_call(['rm', '-rf', settings.STATIC_ROOT])
23	        subprocess.check_call(['npm', 'run', 'gulp', '--', 'build'])

--------------------------------------------------
>> Issue: [B607:start_process_with_partial_path] Starting a process with a partial executable path
   Severity: Low   Confidence: High
   Location: ./selenium_tests/utils.py:22
21	    if not _static_assets_built:
22	        subprocess.check_call(['rm', '-rf', settings.STATIC_ROOT])
23	        subprocess.check_call(['npm', 'run', 'gulp', '--', 'build'])

--------------------------------------------------
>> Issue: [B603:subprocess_without_shell_equals_true] subprocess call - check for execution of untrusted input.
   Severity: Low   Confidence: High
   Location: ./selenium_tests/utils.py:23
22	        subprocess.check_call(['rm', '-rf', settings.STATIC_ROOT])
23	        subprocess.check_call(['npm', 'run', 'gulp', '--', 'build'])
24	        management.call_command('collectstatic', '--noinput', '--link',

--------------------------------------------------
>> Issue: [B607:start_process_with_partial_path] Starting a process with a partial executable path
   Severity: Low   Confidence: High
   Location: ./selenium_tests/utils.py:23
22	        subprocess.check_call(['rm', '-rf', settings.STATIC_ROOT])
23	        subprocess.check_call(['npm', 'run', 'gulp', '--', 'build'])
24	        management.call_command('collectstatic', '--noinput', '--link',

--------------------------------------------------

Code scanned:
	Total lines of code: 3788
	Total lines skipped (#nosec): 0

Run metrics:
	Total issues (by severity):
		Undefined: 0.0
		Low: 17.0
		Medium: 0.0
		High: 2.0
	Total issues (by confidence):
		Undefined: 0.0
		Low: 0.0
		Medium: 2.0
		High: 17.0
Files skipped (0):
```
</details>

File: requirements.txt (hunk 2 above)
Comment (toolness) [inline requirements.txt]: My only concern about this is just the fact that a new bandit release won't trigger a rebuild of a docker container image... and telling docker to "act like `requirements.txt` has changed even though it hasn't" is actually kind of tricky... so from a docker development ergonomics perspective, it'd actually be easier to keep this pinned to a specific version number, even though it doesn't matter much from a security/safety standpoint... 😣 

I wonder if we could get away with just running `pip list  --outdated` every once in a while and update it manually, without requiring a PR to make the change?

**Comment by toolness** [comment]
This looks good to me!  I think we can revisit the issue about the [version range](https://github.com/18F/calc/pull/447#discussion_r72261108) later if it ends up being a problem or something. For now this PR has sat in waiting for TOO LONG and it MUST BE MERGED!

File: requirements.txt (hunk 2 above)
Comment (jseppi) [inline requirements.txt]: Hopefully if a new release ends up finding a security issue, then the Travis build will let you know and you can manually rebuild your dockers at that point. It doesn't actually seem like new bandit releases have happened all that frequently.
